### DISCUSSION TITLE
tracing_appender did not work as global?

### DISCUSSION THREAD

**Discussion by Xujin279** [tokio-rs/tracing#2210] [Q&A] [answered]
hello, I suffered from a problem that tracing_append did not work as global. 
I register a subscriber in log_init(), and then call log_init() in main function. 
![image](https://user-images.githubusercontent.com/53472805/177971517-c71f4006-7e1e-4b87-9fc7-9e8565ddf432.png)
app.log get only one record(register done), but all log informantions print to the terminal.
![image](https://user-images.githubusercontent.com/53472805/177973382-8cb11af0-59e9-4701-8e44-29d1dd2049c6.png)

Did I miss any details？thanks a lot.

**Comment by davidbarsky** [comment, marked as answer]
You're currently dropping `tracing_appender`'s `_guard` at the end of `log_init`. Instead, it _must_ be returned from the `log_init` function to `fn main` (or whatever function lives as long as the program does), as dropping the guard will close the file writer. For additional details, see the documentation on [`WorkerGuard`](https://docs.rs/tracing-appender/latest/tracing_appender/non_blocking/struct.WorkerGuard.html). Once you do that, you should see things logged to the file.

**Reply by Xujin279** [reply]
Thanks for reply, fellowing your suggestion, it works. Thanks a lot again!
